packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
 // This should be exported from soql-builder-ui
 export interface SoqlEditorEvent {
   type: string;
-  message?: string;
+  payload?: string | string[] | ToolingModelJson;
 }
 
 // This should be shared with soql-builder-ui
 export enum MessageType {
-  ACTIVATED = 'activated',
-  QUERY = 'query',
+  UI_ACTIVATED = 'ui_activated',
+  UI_SOQL_CHANGED = 'ui_soql_changed',
   SOBJECT_METADATA_REQUEST = 'sobject_metadata_request',
   SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
   SOBJECTS_REQUEST = 'sobjects_request',
   SOBJECTS_RESPONSE = 'sobjects_response',
-  UPDATE = 'update'
+  TEXT_SOQL_CHANGED = 'text_soql_changed'
+}
+
+// TODO: move to shared module
+async function getConnection(): Promise<Connection> {
+  const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(true);
+  if (!usernameOrAlias) {
+    // TODO: NLS
+    throw new Error(
+      'No default org is set. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.'
```

Comment:
Just noting that we have a separate work item for nls.  "W-8048923 - [soql-builder] i18n for customer-facing strings"